main/integration-tests/support/setup.js

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * This class serves two main purposes:
+ * - Contains the logic for the setup that is applicable to all test suites.
+ * - The entry point to gain access to the default admin session or create other client sessions
+ *
+ * In your rests, you can simply use const setup = await runSetup(); to gain access to an instance of
+ * this class.
+ */
+class Setup {
+  constructor() {
+    this.sessions = [];
+  }
+
+  async init() {
+    // 1 - Read the global __settings__ object and turn it into the settings object, the __settings__ was
+    //     made available in the globals by jest.config.js file
+    // 2 - Use the adminIdToken from settings, to create the default admin session
+
+    // eslint-disable-next-line no-undef
+    const settingsInGlobal = __settings__;
+    if (_.isEmpty(settingsInGlobal)) {
+      throw Error(
+        'No settings variable was found in the globals. Check jest.config.js to ensure that the logic that populates the __settings__ is executed.',
+      );
+    }
+
+    // eslint-disable-next-line no-undef
+    this.settings = new Settings(__settings__);
+    this.apiEndpoint = this.settings.get('apiEndpoint');
+
+    // value generators
+    this.gen = await getGenerators({ setup: this });
+  }
+
+  async defaultAdminSession() {
+    // Only create a new client session if we haven't done that already
+    if (this.defaultAdminSessionInstance) return this.defaultAdminSessionInstance;
+
+    const idToken = this.settings.get('adminIdToken');
+    // In the future, we can check if the token expired and if so, we can create a new one
+    const session = await getClientSession({ idToken, setup: this });
+    this.sessions.push(session);
+    this.defaultAdminSessionInstance = session;
+
+    return session;
+  }
+
+  async createAdminSession() {
+    const adminSession = await this.defaultAdminSession();
+    const username = this.gen.username({ prefix: 'test-admin' });
+    const password = this.gen.password();
+
+    await adminSession.resources.users.create({
+      username,
+      email: username,
+      password,
+      isAdmin: true,
+      userRole: 'admin',
+    });
+
+    const idToken = await getIdToken({ username, password, apiEndpoint: this.apiEndpoint });
+    const session = await getClientSession({ idToken, setup: this });
+    this.sessions.push(session);
+
+    return session;
+  }
+
+  async createResearcherSession() {
+    const adminSession = await this.defaultAdminSession();
+    const username = this.gen.username();
+    const password = this.gen.password();
+
+    await adminSession.resources.users.create({
+      username,
+      email: username,
+      password,
+      userRole: 'researcher',
+    });
+
+    const idToken = await getIdToken({ username, password, apiEndpoint: this.apiEndpoint });
+    const session = await getClientSession({ idToken, setup: this });
+    this.sessions.push(session);
+
+    return session;
+  }
+
+  async createAnonymousSession() {
```

Comment:
What does anonymous session mean?

Reply:
Similar to if you use Postman and don't provide the 'Authorization' header.  For example, you can hit the public auth provider configs APIs without needing an Authorization header